discovery-provider/src/tasks/cache_user_balance.py

```diff
@@ -150,14 +187,17 @@ def refresh_user_ids(
         session.commit()
 
         # Remove the fetched balances from Redis set
-        to_remove = [user.user_id for user in needs_refresh]
         logger.info(
             f"cache_user_balance.py | Got balances for {len(user_query)} users, removing from Redis."
         )
-        if to_remove:
-            redis.srem(REDIS_PREFIX, *to_remove)
+        if lazy_refresh_user_ids:
+            redis.srem(LAZY_REFRESH_REDIS_PREFIX, *lazy_refresh_user_ids)
+            # Add the count of the balances
+            redis.incrby(REDIS_ETH_BALANCE_COUNTER_KEY, len(lazy_refresh_user_ids))
+        if immediate_refresh_user_ids:
+            redis.srem(IMMEDIATE_REFRESH_REDIS_PREFIX, *immediate_refresh_user_ids)
             # Add the count of the balances
-            redis.incrby(REDIS_ETH_BALANCE_COUNTER_KEY, len(to_remove))
+            redis.incrby(REDIS_ETH_BALANCE_COUNTER_KEY, len(immediate_refresh_user_ids))
```

Comment:
what is this "REDIS_ETH_BALANCE_COUNTER_KEY" for? if accuracy of it is super important then we should probs dedupe the lazy and immediate user ids before incrementing the redis value

Reply:
idk what this is even used for 🤷